app/controllers/api/v1/moves_controller.rb

```diff
@@ -4,9 +4,14 @@ module Api
   module V1
     class MovesController < ApiController
       def index
-        import_moves_from_nomis
-        moves = Moves::Finder.new(filter_params).call
-        paginate moves, include: MoveSerializer::INCLUDED_ATTRIBUTES
+        moves_params = Moves::ParamsValidator.new(params[:filter])
+        if moves_params.valid?
+          import_moves_from_nomis
+          moves = Moves::Finder.new(filter_params).call
+          paginate moves, include: MoveSerializer::INCLUDED_ATTRIBUTES
+        else
+          render json: { error: moves_params.errors }
```

Comment:
I think there should be a non-200 status code in here somewhere

Reply:
Also can you adapt this to fit with the JSONAPI spec - https://jsonapi.org/format/#errors ?

I think there are a couple of examples elsewhere in the code. Maybe you can reuse https://github.com/ministryofjustice/hmpps-book-secure-move-api/blob/master/app/controllers/api_controller.rb#L40 ?

Reply:
You might need to change the Swagger docs depending on whether this case is covered by an existing scenario.